src/flow.py

```diff
@@ -601,6 +602,8 @@ def process_notifications(self, timeout_secs=0.05, sid=0):
                 break
             try:
                 session.consume_notification(timeout_secs)
+            except KeyboardInterrupt:
+                self._loop_process_notifications.clear()
             except:
```

Comment:
Good catch on identifying the problem.  Another approach to a solution would be to change the `except:` line in the original to read `except Exception:`.  `except:` catches everything, but `Exception` is lower in the class hierarchy, and only catches non-system exceptions.  Both `Exception` and `KeyBoardInterrupt` inherit from `BaseException`.

https://docs.python.org/2/library/exceptions.html#exceptions.KeyboardInterrupt